systems/plants/@Manipulator/Manipulator.m

```diff
@@ -497,9 +497,13 @@ function num_contacts(obj)
       con = BoundingBoxConstraint(jl_min,jl_max);
       con = setName(con,cellfun(@(a) [a,'_limit'],obj.getPositionFrame.coordinates,'UniformOutput',false));
       if isempty(obj.joint_limit_constraint_id)
-        [obj,id] = addStateConstraint(obj,con,1:obj.getNumPositions);
-        obj.joint_limit_constraint_id = id;
+        if any(jl_min~=-inf) || any(jl_max~=inf)
```

Comment:
This is more specific, right?  (What if someone set the lower limit to +inf?  Hopefully just on accident...)



> On Apr 27, 2015, at 8:52 AM, Hongkai Dai <notifications@github.com> wrote:
> 
> In systems/plants/@Manipulator/Manipulator.m:
> 
> > @@ -497,9 +497,13 @@ function num_contacts(obj)
> >        con = BoundingBoxConstraint(jl_min,jl_max);
> >        con = setName(con,cellfun(@(a) [a,'_limit'],obj.getPositionFrame.coordinates,'UniformOutput',false));
> >        if isempty(obj.joint_limit_constraint_id)
> > -        [obj,id] = addStateConstraint(obj,con,1:obj.getNumPositions);
> > -        obj.joint_limit_constraint_id = id;
> > +        if any(jl_min~=-inf) || any(jl_max~=inf)
> use isinf instead?
> 
> —
> Reply to this email directly or view it on GitHub.
>